compiler/src/lntoamm/Microstatement.ts

```diff
@@ -255,17 +255,44 @@ ${varAst.getText()} on line ${varAst.start.line}:${varAst.start.column}`)
           const lookup = microstatements[microstatements.length - 1]
           // TODO: Map support, which requires figuring out if the outer memory object is an array
           // or a map.
-          if (lookup.outputType.typename !== 'int64') {
-            throw new Error(`${segment.getText()} cannot be used in an array lookup as it is not an int64
+          if (lookup.outputType.typename === 'int64') {
```

Comment:
will this affect aliasing to `int` at all?

Reply:
No, the microstatements don't use the aliases.